### ISSUE TITLE
FYI: Arbitrary python command running during xkeysnail startup

### ISSUE THREAD

**Issue by RedBearAK** [rbreaves/kinto#613]
Just to let you know, I tagged you on a new issue I started over on `mooz/xkeysnail` about a strange thing I encountered trying to run an arbitrary python command along with, or instead of, `launch()`; or as part of a macro with or without `launch()` or other keystrokes. 

I did `import time` and then inserted a `time.sleep(5)` in a macro, like this: 

```
    K("RC-Shift-M-l"):      [time.sleep(5), K("Enter"), K("t"), K("e"), K("x"), K("t"), K("Enter")],
```

It was only when running my `xkeysnail` startup script from within a terminal that I was able to notice that this actually has an effect. The startup of `xkeysnail` itself suddenly paused for 5 seconds, and then resumed. 

In other words, the arbitrary python command "works", but is being run when the config file is parsed rather than when I trigger the keyboard shortcut. When I trigger the actual shortcut macro, that command is either being ignored/skipped somehow, or it is actually running but just in some other context (like the terminal where I ran the startup script) where it has no impact on the GUI environment. 

I'm hoping someone over there has some insight into why this is happening. 

EDIT: Also works with just `sleep()` rather than `time.sleep()`. Still pauses the startup rather than delaying the triggered macro.

**Comment by RedBearAK** [comment]
@rbreaves 

Lenbok already came through with an excellent description of what's happening, and an example that actually works. I see a lot of potential in this, even if it's all done right in the config file without modifying `xkeysnail` at all.

**Comment by rbreaves** [comment]
This is actually one reason I have hope that xkeysnail will still be the better remapper to have used then the newer xremap and keyd ones.. I think it's Python language makes it easier to extend and pull in other libraries too. Although looking at keyd it looks like it is mostly C with some Python. Ironic.. Kinto started out 100% C, w/ yea some bash & xkb usage still.

Xremap is all rust & know literally nothing of Rust.

**Comment by RedBearAK** [comment]
@rbreaves 

Yeah, I'm beginning to get a sense of just how fluid and extensible python really is. Lenbok's instantly working example was quite interesting. 

The implication is that someone could rework the "callable" command interpreter so that it could work with multiple sequential callables, like when you stick a bunch of `K()` key strokes into a macro, but right now it just doesn't work that way, which is why the example in the config file doesn't work as expected. Seems like it should be relatively simple to expand those capabilities and make it work as it was actually intended and... basically it could be a lot closer to AHK with just a few changes. If I knew that the heck I was doing with python.

**Comment by RedBearAK** [comment]
Resolved enough to close this.
